discretisedfield/region.py

```diff
@@ -608,6 +748,81 @@ def __truediv__(self, other):
         """
         return self * other ** (-1)
 
+    def allclose(
+        self,
+        other,
+        atol=None,
+        rtol=None,
+    ):
+        r"""Check if two regions are close.
+
+        Two regions are considered to be equal if they have the same minimum
+        and maximum coordinate points: :math:`\mathbf{p}^\text{max}_1 =
+        \mathbf{p}^\text{max}_2` and :math:`\mathbf{p}^\text{min}_1 =
+        \mathbf{p}^\text{min}_2` within a tolerance.
+
+        Parameters
+        ----------
+        other : discretisedfield.Region
+
+            Second operand.
+
+        atol : numbers.Number, optional
+
+            Absolute tolerance. If ``None``, the default value is
+            the smallest edge length of the region multipled by
+            the tolerance factor.
+
+        rtol : numbers.Number, optional
+
+            Relative tolerance. If ``None``, the default value is
+            the smallest edge length of the region multipled by
+            the tolerance factor.
+
+        Returns
+        -------
+        bool
+
+            ``True`` if two regions are equal (within floating-point accuracy) and
+            ``False`` otherwise.
+
+        Examples
+        --------
+        1. Usage of relational operator ``==``.
+
+        >>> import discretisedfield as df
+        ...
+        >>> region1 = df.Region(p1=(0, 0, 0), p2=(5, 5, 5))
+        >>> region2 = df.Region(p1=(0.0, 0, 0), p2=(5.0, 5, 5))
+        >>> region3 = df.Region(p1=(1, 1, 1), p2=(5, 5, 5))
+        ...
+        >>> region1.allclose(region2)
+        True
+        >>> region1.allclose(region2)
+        False
+        >>> region1.allclose(region3)
+        False
+        >>> region1.allclose(region3)
+        True
+
+        """
+        if isinstance(other, self.__class__):
+            if atol is None:
+                atol = np.min(self.edges) * self.tolerance_factor
+            elif not isinstance(atol, numbers.Number):
+                raise TypeError(f"{type(atol)=} is not a number.")
+
+            if rtol is None:
+                rtol = np.min(self.edges) * self.tolerance_factor
+            elif not isinstance(rtol, numbers.Number):
+                raise TypeError(f"{type(rtol)=} is not a number.")
+
+            return np.allclose(
+                self.pmin, other.pmin, atol=atol, rtol=rtol
+            ) and np.allclose(self.pmax, other.pmax, atol=atol, rtol=rtol)
+
+        return False
```

Comment:
Should we not raise an error if `other` is not a `Region` object?

Reply:
I agree, for `allclose` a `TypeError` would be better than returning False. I'll change it. Do you agree @samjrholt ?